op-service/sources/engine_client.go

```diff
@@ -100,28 +95,15 @@ func (s *EngineAPIClient) ForkchoiceUpdate(ctx context.Context, fc *eth.Forkchoi
 	var result eth.ForkchoiceUpdatedResult
 	method := s.evp.ForkchoiceUpdatedVersion(attributes)
 	err := s.RPC.CallContext(fcCtx, &result, string(method), fc, attributes)
-	if err == nil {
-		tlog.Trace("Shared forkchoice-updated signal")
-		if attributes != nil { // block building is optional, we only get a payload ID if we are building a block
-			tlog.Trace("Received payload id", "payloadId", result.PayloadID)
-		}
-		return &result, nil
-	} else {
+	if err != nil {
 		llog.Warn("Failed to share forkchoice-updated signal", "err", err)
-		if rpcErr, ok := err.(rpc.Error); ok {
-			code := eth.ErrorCode(rpcErr.ErrorCode())
-			switch code {
-			case eth.InvalidParams, eth.InvalidForkchoiceState, eth.InvalidPayloadAttributes:
```

Comment:
I believe that `InvalidParams` (generic RPC error), will result in a temporary-error loop, but that's fine, since it implies the op-node/op-program speaks a different RPC than the engine, and stalling/aborting is fine in that case. It wouldn't happen during a correctly set up CL / EL combination.

Reply:
yep, it's just the operational difference between a retry loop vs an instant crash